app/services/candidate_interface/can_add_course_choice.rb

```diff
@@ -0,0 +1,16 @@
+module CandidateInterface
+  class CanAddCourseChoice
```

Comment:
Thinking about the interface here - the question of 'can I add courses to this application form?' is much broader than the end of cycle logic implemented in the class body. For example, we prevent adding courses when the max number of choices has been reached.

I wonder if something like `EndOfCyclePolicy.can_add_course_choice?(application_form)` would be clearer? That way, at each call site it's obvious this check relates to end of cycle concerns.

Reply:
Thanks Malcolm, that is a really good suggestion. I was struggling to come with a good name for this 👍